File: server/src/main/java/com/bettercallxiaojin/home/pojo/entity/Comment.java
package com.bettercallxiaojin.home.pojo.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Comment {
    private String id;
    private String userId;
    private String postId;
    private String content;
    private Integer status;
    private Integer likeCount;
    private Integer replyCount;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: web/src/models/comment.ts
interface SimpleUserVO {
  id: string;
  username: string;
  icon: string;
}

// 请求类型
export interface AddCommentRequest {
  postId: string;
  content: string;
}

export interface CommentRequest {
  id: string;
  content: string;
}

// 响应数据类型
export interface CommentData {
  id: string;
  userVO: SimpleUserVO;
  postId: string;
  content: string;
  status: number;
  likeCount: number;
  replyCount: number;
  isLike: boolean;
  isCreator: boolean;
  createdAt: string;
  updatedAt: string;
}